### PR TITLE
Add glossary terms from prod-dev repo glossary

### PR DESCRIPTION
**Pull request by tannerwelsh**
Applicant, Cohort, Partner

### CODE REVIEW

**Comment by tannerwelsh** [comment]
@shereefb not sure if you want/need to review small changes like these, but here's the PR anyway.

FYI this is part my work creating a central glossary that can be referenced elsewhere, like this: https://github.com/LearnersGuild/product-development/pull/3

File: GLOSSARY.md [hunk 1]
```diff
@@ -11,13 +17,16 @@ The core is the central organizational body of Learners Guild. The core is respo
...
 # Crew
-Learners are organized into crews. Crews are the atomic level of group organization in Learners Guild. Each crew has a defined purpose, domain, and objectives.
+Learners are organized into crews. Crews are the atomic level of group organization in Learners Guild, designed to provide an accountability framework for their component learners. Each crew has a defined purpose, domain, and objectives. Generally crews are composed of 12 learners.
 
 # Guide
 Guides are learners who may or may not be actively working in a particular craft but have nonetheless achieved a high level of expertise in that craft. Their primary responsibility is to guide, mentor, and otherwise assist apprentices and practitioners in their learning.
 
 # Learner
-Everyone who works for or with Learners Guild is a learner. This includes the staff of the core as well as everyone involved with a guild chapter.
+Everyone who works for or with Learners Guild is a learner. This includes the staff of the core (partners) as well as everyone involved with a guild chapter.
```
Comment (shereefb) [inline GLOSSARY.md]: I understand the spirit of this, but it's confusing.
I think it's easier to think of partners and learners as non overlapping. 
We (partners) now have no word that describes who we are serving.

File: GLOSSARY.md [hunk 2]
```diff
@@ -1,9 +1,15 @@
+# Applicant
+An Applicant is a potential learner who hasn't yet completed the admissions process, but has indicated interest in joining the guild.
```
Comment (shereefb) [inline GLOSSARY.md]: What word are we using from someone who is interested but hasn't yet applied?

**Comment by shereefb** [comment]
We lost the word "mentor": Someone who is actively taking a role in supporting someone else's learning.
For example, an apprentice can and will (after a few months at the guild) start mentoring 'younger' apprentices.

File: GLOSSARY.md (hunk 1 above)
Comment (tannerwelsh) [inline GLOSSARY.md]: Agree, it needs to be clarified.

My thinking behind this is that we serve **apprentices**. People who want to join the guild apply to be apprentices, the first 9 months is their apprenticeship, and so forth.

Anyone in the guild, whether paid partner or apprentice, is a learner, in the same way that a everyone in a guild of carpenters is a carpenter.

File: GLOSSARY.md (hunk 2 above)
Comment (tannerwelsh) [inline GLOSSARY.md]: How about **Prospective Learner**?

**Comment by tannerwelsh** [comment]
@shereefb: although I like the implication of a "Guide" better, I agree that mentor is more explicit in this case. Swapped out with 2ef9ba7bd54ee21eb801b1b690e5c7ff9783e7fb

**Pull request merged**